client/data/followers/remove-follower.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { useMutation, useQueryClient } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import wp from 'calypso/lib/wp';
+
+function useRemoveFollower() {
+	const queryClient = useQueryClient();
+	const { mutate: removeFollower, ...rest } = useMutation(
+		( { siteId, follower, type } ) => {
```

Comment:
All three arguments are mandatory so they could be three arguments:
```js
removeFollower( siteId, type, followerId )
```
The order is from the least specific (site) to the most specific (individual follower ID).

And we only need to pass the `follower.ID`, not the entire object.

Reply:
The `mutate` function returned from `useMutation` expects variables to be passed as object (see [reference](https://react-query.tanstack.com/reference/useMutation)) so we can't pass them as suggested. We could ofc use a custom signature for `removeFollower` which then wraps the original `mutate` fn instead of renaming it as is now.

Reply:
I think a custom signature for `removeFollower` is fine. It denotes that some variables are mandatory.

`useMutation` return value also includes a promise-returning `mutateAsync`. If we're removing `mutate`, we should remove `mutateAsync`, too.

Reply:
Agreed. Either way I think we should not hide `mutate` and `mutateAsync` to make it possible for consumers to use the underlying "low level" method and don't introduce exceptions for certain returned properties.

Reply:
What do you think of https://github.com/Automattic/wp-calypso/pull/50378/commits/9395f5d3a20e8c493d64037da738b1b063c4eeaa?

Reply:
Yes, just adding `removeFollower` to the object without removing anything sounds good. We'll see in future what parts of the `useMutation` return value we really use.